### PR TITLE
refactoring data module

### PR DESCRIPTION
**Pull request by eddiepyang**

### CODE REVIEW

File: yelp_nlp/tests/test_data_module.py [hunk 1]
```diff
@@ -1,13 +1,11 @@
 import pytest
 import pandas as pd
```
Comment (eddiepyang) [inline yelp_nlp/tests/test_data_module.py:2]: this will need to be refactored to seperate modules too

File: yelp_nlp/rnn/config.py [hunk 2]
```diff
@@ -3,6 +3,16 @@
 from torch import nn
 from dataclasses import dataclass
 from yelp_nlp import root
+from logging import NOTSET, DEBUG, INFO, WARN, ERROR, CRITICAL
+
+
+class LogLevels(enum.Enum):
```
Comment (ryan-dayrit) [inline yelp_nlp/rnn/config.py:9]: chicken

File: yelp_nlp/rnn/extract.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import zipfile
+from gensim import corpora
+
+from yelp_nlp.rnn.config import LogLevels
+from yelp_nlp.logging_utils import new_logger, time_decorator
+
+logger = new_logger(LogLevels.debug.value)
+
+
+def load_embeddings(
```
Comment (ryan-dayrit) [inline yelp_nlp/rnn/extract.py:11]: extract_embeddings maybe is better here

File: yelp_nlp/rnn/load.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    def __len__(self):
+        return self.data.__len__()
+
+    def __getitem__(self, i):
+        return torch.tensor(self.data[self.x_labels].iat[i]), torch.tensor(
+            self.data[self.y_labels].iat[i]
+        )
+
+
+def new_train_val_datasets(
```
Comment (ryan-dayrit) [inline yelp_nlp/rnn/load.py:47]: load_dataset?

File: yelp_nlp/rnn/load.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+def split_df(
+    df: pd.DataFrame, test_size: float = 0.25
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+
+    tr_idx, val_idx = train_test_split(df.index.values, test_size=test_size)
+    return df.iloc[tr_idx], df.iloc[val_idx]  # type: ignore
+
+
+@define
+class CorpusDataset(Dataset):
```
Comment (ryan-dayrit) [inline yelp_nlp/rnn/load.py:27]: maybe this belongs in models.py

File: yelp_nlp/rnn/runner.py [hunk 6]
```diff
@@ -1,26 +1,20 @@
-from typing import Tuple
-import pandas as pd
 import os
```
Comment (ryan-dayrit) [inline yelp_nlp/rnn/runner.py:1]: possibly driver.py is better naming

**Comment by ryan-dayrit** [comment]
Please fill in pr template and add task id identifier to pr title. 
Screen Shot 2022-04-18 at 9 21 26 PM

File: yelp_nlp/rnn/load.py (hunk 4 above)
Comment (eddiepyang) [reply yelp_nlp/rnn/load.py]: hmm..as the factory function new.* feels better here

Comment (ryan-dayrit) [reply yelp_nlp/rnn/load.py]: It would be new if your filename was dataset.py

Comment (eddiepyang) [reply yelp_nlp/rnn/load.py]: it's creating new CorpusDataset classes though which inherit from Dataset, maybe new_.*_corpus_datasets is the appropriate name

File: yelp_nlp/rnn/load.py (hunk 5 above)
Comment (eddiepyang) [reply yelp_nlp/rnn/load.py]: Feels better to decouple this from the model class, as potentially there could be multiple models in the file.

File: yelp_nlp/rnn/config.py (hunk 2 above)
Comment (eddiepyang) [reply yelp_nlp/rnn/config.py:9]: image

**Pull request merged**